src/javascript/esutil.ts

```diff
@@ -284,6 +284,33 @@ export function getReturnFromAnnotation(jsdocAnn: jsdoc.Annotation):
   return type;
 }
 
+/**
+ * Examine the body of a function to see if we can infer something about its
+ * return type. This currently only handles the case where a function definitely
+ * returns void.
+ */
+export function inferReturnFromBody(node: babel.Function): {type: string}|
+    undefined {
+  if (node.async === true || node.generator === true) {
+    return undefined;
+  }
+  let returnsVoid = true;
+  estraverse.traverse(node, {
+    enterReturnStatement(statement: babel.ReturnStatement) {
```

Comment:
Can have fewer false negatives if you do something like:

```typescript
enterFunction() {
  return estraverse.VisitorOption.Skip;
}
```

Reply:
Done.